library/bigip_software.py

```diff
@@ -46,64 +40,148 @@
-        will be removed from the system
+      - When C(installed), ensures that the software is uploaded/downloaded 
+        and installed on the system. The device is not, however, 
+        rebooted into the new software.
+      - When C(activated), ensures that the software is uploaded/downloaded, 
+        installed, and the system is rebooted to the new software. 
+      - When C(present), ensures that the software is uploaded/downloaded. 
+      - When C(absent), only the uploaded/downloaded image 
+        will be removed from the system.
     required: false
     default: activated
     choices:
       - absent
       - activated
       - installed
       - present
+    volume:
+    description:
+      - The volume to install the software and, optionally, the hotfix to. This
+        parameter is only required when the C(state) is either C(activated) or
+        C(installed).
+    required: false
   software:
     description:
       - The path to the software (base image) to install. The parameter must be
         provided if the C(state) is either C(installed) or C(activated).
+      - If C(remote_src) is used, this parameter will have to be an 
+        C(unauthenticated) C(HTTP) or C(HTTPS)link to the software image.
+      - C(Caveat): When providing link to the software ISO, 
+        if the ISO name is different than the one listed inside the 
+        C(software_md5) md5sum file. We will change it accordingly when 
+        saving the files on the device. This might lead to ISO names not 
+        matching the links provided in C(software).
     required: false
     aliases:
       - base_image
-  volume:
+  hotfix:
     description:
-      - The volume to install the software and, optionally, the hotfix to. This
-        parameter is only required when the C(state) is either C(activated) or
-        C(installed).
+      - The path to an optional Hotfix to install. This parameter requires that
+        the C(software) parameter be specified or the corresponding software 
+        image exists on the unit.
+      - If C(remote_src) is used, this parameter will have to be an 
+        unauthenticated http or https link to the hotfix image.
+      - C(Caveat): When providing link to the hotfix ISO, if the ISO name 
+        is different than the one listed inside the C(hotfix_md5) md5sum file. 
+        We will change it accordingly while saving the files on the device. 
+        This might lead to ISO names not matching the links provided 
+        in C(hotfix).        
     required: false
+    default: None
+    aliases:
+      - hotfix_image
+  software_md5sum:
+    description:
+      - The link to an MD5 sum file of the remote software ISO image, 
+        it is required when  C(software) parameter is used and C(remote_src) 
+        is selected. 
+      - Parameter only used when and C(state) is: C(installed), C(activated) 
+        or C(present).
+    required: false
+    default: None      
```

Comment:
you can remove all lines that are

"required: false"
"default: None"

this is a new request from upstream vendor, so I've begun making the switch